File: python/Motorcycle.py
from Vehicle import Vehicle


# Motorcycle : jenis motor, kapasitas tangki
class Motorcycle(Vehicle):
    __jenis_motor = ""
    __kapasitas_tangki = ""

    def __init__(self, jenis_motor, kapasitas_tangki, plat, merk, tahun, warna):
        # mengisi kontruktor parent
        super().__init__(plat, merk, tahun, warna)

        self.__jenis_motor = jenis_motor
        self.__kapasitas_tangki = kapasitas_tangki

    # getter setter
    def getJenisMotor(self):
        return self.__jenis_motor

    def setJenisMotor(self, jenis):
        self.__jenis_motor = jenis

    def getKapasitasTangki(self):
        return self.__kapasitas_tangki

    def setKapasitasTangki(self, kapasitas):
        self.__kapasitas_tangki = kapasitas

File: cpp/Motorcycle.cpp
#pragma once
#include <iostream>
#include <string>

#include "Vehicle.cpp"
using namespace std;

class Motorcycle : public Vehicle {
 private:
  string jenis_motor;
  string kapasitas_tangki;

 public:
  Motorcycle() {}
  Motorcycle(string jenis_motor, string kapasitas_tangki) {
    this->jenis_motor = jenis_motor;
    this->kapasitas_tangki = kapasitas_tangki;
  }
  //   overloading baby!
  Motorcycle(string jenis_motor, string kapasitas_tangki, string plat,
             string merk, string tahun, string warna)
      : Vehicle(plat, merk, tahun, warna) {
    this->jenis_motor = jenis_motor;
    this->kapasitas_tangki = kapasitas_tangki;
  }

  // accessor mutator
  string getJenisMotor() { return this->jenis_motor; }
  void setJenisMotor(string jenisMotor) { this->jenis_motor = jenisMotor; }

  string getKapasitasTangki() { return this->kapasitas_tangki; }
  void setKapasitasTangki(string kapasitasTangki) {
    this->kapasitas_tangki = kapasitas_tangki;
  }

  void printDetails() override {
    // kita pake polymorphism
    Vehicle::printDetails();
    cout << "Jenis Motor: " << getJenisMotor() << endl;
    cout << "Kapasitas Tangki: " << getKapasitasTangki() << endl;
  }
  ~Motorcycle() {}
};
